jax/__init__.py

```diff
@@ -174,6 +166,45 @@
 # Also circular dependency.
 from jax._src.array import Shard as Shard
 
+# TODO(jakevdp) update callers and remove these after June 2023
```

Comment:
Consider using the utility/pattern in deprecations.py for this?

Canonical example: https://github.com/google/jax/blob/main/jax/interpreters/pxla.py

One key benefit is then you can locally (e.g., in our internal copy of the JAX) accelerate the deprecations to prevent backsliding when all users we know about are gone.

Reply:
Sounds good, thanks!

Reply:
Done